src/main/java/org/springframework/data/domain/CursorRequest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.domain;
+
+/**
+ * Interface defining a request for a cursor window. Typically used to fetch subsets of query results if the query
+ * results are too large to fit entirely into memory and to be able to resume a cursoring session.
+ *
+ * @author Mark Paluch
+ * @since 3.1
+ * @see OffsetCursorRequest
+ * @see KeysetCursorRequest
+ * @see CursorWindow
+ * @see Pageable#toCursorRequest()
+ */
+public interface CursorRequest {
+
+	/**
+	 * Returns a {@link CursorRequest} instance representing no cursor setup.
+	 *
+	 * @return
+	 */
+	static CursorRequest uncursored() {
+		return Uncursored.INSTANCE;
+	}
+
+	/**
+	 * Returns whether the current {@link CursorRequest} contains cursor information.
+	 *
+	 * @return
+	 */
+	boolean isCursored();
+
+	/**
+	 * Returns whether the current {@link CursorRequest} does not contain cursor information.
+	 *
+	 * @return
+	 */
+	default boolean isUncursored() {
+		return !isCursored();
+	}
+
+	/**
+	 * Returns the number of items to be returned.
+	 *
+	 * @return the number of items of that cursor request.
+	 */
+	int getSize();
+
+	/**
+	 * Creates a new cursor request with the given size.
+	 *
+	 * @param size the cursor result size.
+	 * @return a new {@link CursorRequest} with {@code size} applied.
+	 */
+	CursorRequest withSize(int size);
+
+	/**
+	 * Returns the sorting parameters for this cursor request.
+	 *
+	 * @return
+	 */
+	Sort getSort();
+
+	/**
+	 * Creates a new cursor request with the {@link Sort}.
+	 *
+	 * @param sort the cursor result sort order.
+	 * @return a new {@link CursorRequest} with {@link Sort} applied.
+	 */
+	CursorRequest withSort(Sort sort);
```

Comment:
Why does a request need to know about the sort? I agree that there's value in the ability to change the window size with each request, but changing result order while scrolling through data does not make senes, does it?

Reply:
This was introduced to carry over the final sort from a query method that either defines a sort or accepts a `Sort` argument.